2025/lab/Keys/Week7_Missingness_Robustness.Rmd

```diff
@@ -0,0 +1,551 @@
+
+-   For more information, see John Fox's response to the question here: <https://stats.stackexchange.com/questions/70679/which-variance-inflation-factor-should-i-be-using-textgvif-or-textgvif>
+-   And the linked paper here: <https://www.tandfonline.com/doi/abs/10.1080/01621459.1992.10475190>
+
+### High Multicollinearity
+
+Let's try to now create a model that *will* have high multicollinearity. Work experience and age should be correlated. Without looking at the numbers, you can imagine how someone older would generally have more work experience than someone younger. Let's see if that plays out in the model
+
+```{r high-multicol}
+
+multicol_model <- lm(logincome ~ prestige + educ + workexp + agesq, data = gss)
+
+summary(multicol_model)
+
+# Calculate VIF for the adjusted model
+multicol_vif <- vif(multicol_model)
+multicol_vif <- as.data.frame(multicol_vif)
+
+print(multicol_vif)
+```
+
+You can see here that `workexp` and `agesq` are collinear with each other.
+
+Multicollinearity can present as a problem in modeling because we aim for each predictor to capture unique portions of the variance in $\hat{y}$. Instead, variables that are highly multicollinear are not capturing unique portions of the variance, affecting the standard errors. 
+
+## 5. Bayesian Specific
+
+Predictive performance can be calculated using cross-validation. These approaches are infrequently used in frequentist statistics but are central to modern Bayesian inference.
+
+### Leave-One-Out Cross Validation
+
+One form of cross validation is Leave-One-Out, where we leave out one data point and see how good the model does. **NOTE**: While this can be done on OLS models, the `rstanarm::loo()` function only works on `stan_glm` objects (not `lm`). Because of this, let's create two Bayesian models and see how they do. (Also, you sadly can't use data weights with `loo()`, so we'll create unweighted models.)
+
+```{r bayesmod-create}
+bayes_mod1 <- stan_glm(logincome ~ prestige + self_employed + race + female,
+                       data = gss_comp,  seed = seed,
+                       chains = 1, refresh = 0)
+bayes_mod2 <- stan_glm(logincome ~ prestige + self_employed + race + female * (workexp + workexp^2),
+                       data = gss_comp,  seed = seed,
+                       chains = 1, refresh = 0)
+
+bmods <- list(bayes_mod1, bayes_mod2)
+modelsummary(bmods,
+             statistic = "conf.int", gof_omit = "F|RMSE",
+             output = "huxtable")
+```
+
+Let's run `loo()` on each of our models and then use `loo_compare()` to compare the results.
+
+```{r loo}
+loo1 <- loo(bayes_mod1)
+loo2 <- loo(bayes_mod2)
+loo_compare(loo1, loo2)
+```
+
+The comparison shows that the first model tends to perform worse according to the ELPD metric than model 2.
+
+We can also plot the Pareto k values, which can indicate whether there are data points that may bias the results. Ideally, all values should be less than 0.7.
+
+```{r}
+plot(loo1)
+plot(loo2)
+```
+
```

Comment:
@brenthoagland I briefly discuss this in lecture but only show `loo` because k-fold is not really needed for regression. Please remove this last part.